src/asm_syntax.hpp

```diff
@@ -235,11 +235,31 @@ struct Packet {
     constexpr bool operator==(const Packet&) const = default;
 };
 
-/// Special instruction for incrementing values inside shared memory.
-struct LockAdd {
+/// Special instruction for atomically updating values inside shared memory.
```

Comment:
Maybe it's worth explaining here that the analysis just treats them as 3 consecutive operations, and the atomicity is insignificant.

Reply:
Added comment